### PR TITLE
Creating s2a directory

### PR DESCRIPTION
**Pull request by dmbm0830**
<!--
!!!ATTENTION!!!

If you are fixing *any* crash or *any* potential security issue, *do not*
open a pull request in this repo. Please report the issue via emailing
envoy-security@googlegroups.com where the issue will be triaged appropriately.
Thank you in advance for helping to keep Envoy secure.

!!!ATTENTION!!!

### CODE REVIEW

File: source/extensions/transport_sockets/s2a/BUILD [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "grpc_tsi.h",
+    ],
+    external_deps = [
+        "grpc",
+    ],
+    visibility = ["//visibility:private"],
+    deps = [
+        "//source/common/common:c_smart_ptr_lib",
+    ],
+)
```
Comment (matthewstevenson88) [inline source/extensions/transport_sockets/s2a/BUILD]: nit: please add a new line at the end of the file.

File: source/extensions/transport_sockets/s2a/grpc_tsi.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+using CFrameProtectorPtr =
+    CSmartPtr<tsi_zero_copy_grpc_protector, tsi_zero_copy_grpc_protector_destroy>;
+
+using CHandshakerResultPtr = CSmartPtr<tsi_handshaker_result, tsi_handshaker_result_destroy>;
+using CHandshakerPtr = CSmartPtr<tsi_handshaker, tsi_handshaker_destroy>;
+
+} // namespace S2A
+} // namespace TransportSockets
+} // namespace Extensions
+} // namespace Envoy
```
Comment (matthewstevenson88) [inline source/extensions/transport_sockets/s2a/grpc_tsi.h]: nit: please add a new line at the end of the file.

File: source/extensions/transport_sockets/s2a/grpc_tsi.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+#include "source/common/common/c_smart_ptr.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace S2A {
+
+using CFrameProtectorPtr =
+    CSmartPtr<tsi_zero_copy_grpc_protector, tsi_zero_copy_grpc_protector_destroy>;
+
```
Comment (matthewstevenson88) [inline source/extensions/transport_sockets/s2a/grpc_tsi.h]: nit: can we remove the empty line between the using declarations?

**Comment by matthewstevenson88** [comment]
Generally LGTM, just 3 micro-nits. :)

**Review by matthewstevenson88 — APPROVED** [review]

**Review by tavishvaidya — APPROVED** [review]

**Pull request merged**